management-api-agent-common/src/main/java/io/k8ssandra/metrics/builder/CassandraMetricNameParser.java

```diff
@@ -141,8 +141,20 @@ public void replace(String dropwizardName, CassandraMetricDefinition metricDefin
           if (relabel.getTargetLabel().equals(RelabelSpec.METRIC_NAME_LABELNAME)) {
             metricDefinition.setMetricName(output);
           } else {
-            metricDefinition.getLabelNames().add(relabel.getTargetLabel());
-            metricDefinition.getLabelValues().add(output);
+            // Not the most effective way of doing this (map would be better - but perf is not an
+            // issue since this isn't done in the hotpath)
+            boolean replaced = false;
```

Comment:
I had a play with this. A map might have made life easier, but it isn't a big deal. 

Maybe consider doing something with `.stream()` for readability. There are some methods like `match` and `filter` that may do the job for you. It isn't a big deal either way IMO.